overlord/snapstate/snapstate.go

```diff
@@ -1063,6 +1064,44 @@ func InstallWithDeviceContext(ctx context.Context, st *state.State, name string,
 	return doInstall(st, &snapst, snapsup, 0, fromChange, nil)
 }
 
+func InstallPathMany(st *state.State, sideInfos []*snap.SideInfo, paths, instanceNames, channels []string, flags Flags) ([]*state.TaskSet, []*snap.Info, error) {
```

Comment:
This probably can take a bit less arguments to align more with what UpdateMany/InstallMany accept. We don't let people control as much as with single snap operations in the many cases.